File: benchmarks/raytracer/raytracer-cpp/src/sphere.cpp
#pragma once

#include "vector.cpp"
#include "material.cpp"

/*
 * Class holding data about a sphere
 */
class Sphere {

	public:
		// the position of the center of this sphere 
		Vector center;
		// the radius of this sphere
		float radius;
		// the surface material
		Material material;

		// dummy sphere
		static Sphere NONE;

		static Sphere create(const Vector *center, const float radius, const Material *material) {
			Sphere sphere;
			sphere.center = *center;
			sphere.radius = radius;
			sphere.material = *material;
			return sphere;
		}
};

Sphere Sphere::NONE = Sphere::create(&Vector::ZERO, 0.0, &Material::NONE);

File: benchmarks/raytracer/raytracer-js/src/sphere.js
import { Vector } from './vector.js';
import { Material } from './material.js';

/**
 * Class holding data about a sphere
 */
export class Sphere {

	// Dummy sphere
	static NONE = new Sphere(Vector.ZERO, 0, Material.NONE);

	constructor(center, radius, material) {
		// the position (vector) of the center of this sphere 
		this.center = center;
		// the radius of this sphere
		this.radius = radius;
		// the surface material
		this.material = material;
	}


}